beverage_distributor/data/ir_actions_server.xml

```diff
@@ -27,13 +28,15 @@ if record.x_unit_sale_product:
         uom_id = env['uom.uom'].search([('relative_factor', '=', quant), ('relative_uom_id', '=', record.x_unit_sale_product.uom_id.id)], limit=1)
         if not uom_id:
             uom_id = env['uom.uom'].create({'relative_factor': quant, 'relative_uom_id': record.x_unit_sale_product.uom_id.id, 'name': f"Pack of {quant} x {record.x_unit_sale_product.uom_id.name}"})
-        env['stock.warehouse.orderpoint'].create({
-            'product_id': record.x_unit_sale_product.product_variant_id.id,
-            'x_parent_product': record.id,
-            'replenishment_uom_id': uom_id.id,
-            'route_id': env.ref('mrp.route_warehouse0_manufacture').id,
-            'location_id': env.ref('stock.stock_location_stock').id,
-        })
+        for wh in env['stock.warehouse'].search([('lot_stock_id', '!=', False)]):
+            env['stock.warehouse.orderpoint'].create({
+                'product_id': record.x_unit_sale_product.product_variant_id.id,
+                'x_parent_product': record.id,
+                'replenishment_uom_id': uom_id.id,
+                'company_id': wh.company_id.id,
+                'route_id': env.ref('mrp.route_warehouse0_manufacture').id,
+                'location_id': wh.lot_stock_id.id,
+            })
```

Comment:
```suggestion
warehouses = env['stock.warehouse'].search([('lot_stock_id', '!=', False)])
orderpoints_to_create = []
route = env.ref('mrp.route_warehouse0_manufacture', raise_if_not_found=False)

for wh in warehouses:
    orderpoints_to_create.append({
        'product_id': record.x_unit_sale_product.product_variant_id.id,
        'x_parent_product': record.id,
        'replenishment_uom_id': uom_id.id,
        'company_id': wh.company_id.id,
        'route_id': route.id,
        'location_id': wh.lot_stock_id.id,
    })

if orderpoints_to_create:
    env['stock.warehouse.orderpoint'].create(orderpoints_to_create)
```
Just to create in batch. Also, we can add a check if record.x_unit_sale_product.product_variant_id is not set.